### PR TITLE
[FIXED JENKINS-53576] - Improve JCASC compatibility

### PR DESCRIPTION
**Pull request by jonbrohauge**
Add getCustomDomain() to improve JCASC compatibility.

### CODE REVIEW

**Comment by jonbrohauge** [comment]
Ping @fbelzunc

File: src/main/java/hudson/plugins/active_directory/ActiveDirectorySecurityRealm.java [hunk 1]
```diff
@@ -311,6 +311,11 @@ public Integer getTtl() {
         return environmentProperties;
     }
 
+    @Restricted(NoExternalUse.class)
+    public boolean getCustomDomain() {
```
Comment (jvz) [inline src/main/java/hudson/plugins/active_directory/ActiveDirectorySecurityRealm.java]: Wouldn't this make more sense named as `isCustomDomain()`?

Comment (jonbrohauge) [reply src/main/java/hudson/plugins/active_directory/ActiveDirectorySecurityRealm.java]: Just trying to make a least intrusive change to make this plugin work with [Configuration as code](https://github.com/jenkinsci/configuration-as-code-plugin)

**Comment by markltbaker** [comment]
@fbelzunc can the AD plugin get released so we can use it with Jenkins config as a code? Thanks!

**Comment by fbelzunc** [comment]
@markltbaker Yes, I am doing it.

**Pull request merged**